react-web/src/pages/userProfile/UserProfile.tsx

```diff
@@ -74,129 +125,314 @@ const UserProfile = () => {
-            disabled={!isEdit}
-            type="text"
-            required={true}
-            disablefocus="true"
-            {...form.register("owner")}
-          />
-          <Input
-            label="dApp Repository"
-            disabled={!isEdit}
-            type="text"
-            required={true}
-            disablefocus="true"
-            {...form.register("repo")}
-          />
-          <Input
-            label="dApp Version"
-            type="text"
-            id="version"
-            required={true}
-            disabled={!isEdit}
-            disablefocus="true"
-            {...form.register("version")}
-          />
-          <Input
-            label="Authors"
-            type="text"
-            id="authors"
-            disabled={!isEdit}
-            disablefocus="true"
-            {...form.register("authors")}
-          />
-          <Input
-            label="Contacts"
-            type="text"
-            id="contacts"
-            disabled={!isEdit}
-            disablefocus="true"
-            {...form.register("contacts")}
-          />
-          <Input
-            label="Vendor"
-            type="text"
-            id="vendor"
-            disablefocus="true"
-            disabled={!isEdit}
-            {...form.register("vendor")}
-          />
-          <Input
-            label="Website"
-            type="text"
-            id="website"
-            disabled={!isEdit}
-            disablefocus="true"
-            {...form.register("website")}
-          />
-          <Input
-            label="LinkedIn Url"
-            type="text"
-            id="linkedIn"
-            disablefocus="true"
-            disabled={!isEdit}
-            {...form.register("linkedin")}
-          />
-          <Input
-            label="Twitter Url"
-            type="text"
-            id="twitter"
-            disabled={!isEdit}
-            disablefocus="true"
-            {...form.register("twitter")}
-          />
-          <div className="button-wrapper">
-            {!isEdit ? (
-              <Button
-                type="button"
-                buttonLabel={"Edit"}
-                onClick={(e) => {
-                  setIsEdit(!isEdit);
+    <>
+      <div id="profileContainer">
+        <div>
+          <Form form={form} onSubmit={formHandler}>
+            <Input
+              label="dApp Name"
+              type="text"
+              id="name"
+              required={true}
+              disabled={!isEdit}
+              disablefocus="true"
+              {...form.register("name")}
+            />
+            <div className="input-wrapper">
+              <div
+                className={`input ${activeOwner ? "active" : ""} ${
+                  !isEdit ? "disabled" : ""
+                }`}
+                onClick={(_) => {
+                  setActiveOwner(true);
+                  setFocussedOwnerRepo(true)
+                  document.getElementById("owner" || "")?.focus();
```

Comment:
shouldn't we use component refs instead of DOM ids? 

I know it is an old example with `class` components but still the case imo
https://www.javascriptstuff.com/use-refs-not-ids/